doc/concepts/moveit_task_constructor/generating_stages.rst

```diff
@@ -0,0 +1,160 @@
+.. _Generating Stages:
+
+#################
+Generating Stages
+#################
+
+Generator stages get no input from adjacent stages. They compute results and pass them to adjacent stages.
+
+MTC provides the following generator stages:
+
+* ``CurrentState``
+
+* ``FixedState``
+
+* Monitoring Generators - ``GeneratePose``, ``GenerateGraspPose``, ``GeneratePlacePose`` and ``GenerateRandomPose``.
+
+CurrentState
+-------------
+| The ``CurrentState`` stage fetches the current PlanningScene via the ``get_planning_scene`` service.
+| This stage is often used at the beginning of the MTC task pipeline to set the start state from the current robot state.
+
+Example code
+
+.. code-block:: c++
+
+  auto current_state = std::make_unique<moveit::task_constructor::stages::CurrentState>("current_state");
+
+`API doc for CurrentState <https://ros-planning.github.io/moveit_task_constructor/_static/classmoveit_1_1task__constructor_1_1stages_1_1CurrentState.html>`_.
+
+FixedState
+----------
+
+| The ``FixedState`` stage spawns a pre-defined PlanningScene State.
```

Comment:
Might be worth adding a sentence saying that this stage is useful for planning from an intended future state, enabling things like simultaneous planning and execution.